python_modules/dagster/dagster/_core/definitions/asset_checks.py

```diff
@@ -127,3 +143,70 @@ def get_attributes_dict(self) -> Dict[str, Any]:
             specs=self._specs,
             input_output_props=self._input_output_props,
         )
+
+
+@experimental
+def build_blocking_asset_check(
+    asset_def: "AssetsDefinition",
+    checks: Sequence[AssetChecksDefinition],
+) -> "AssetsDefinition":
+    from dagster import AssetIn, In, OpExecutionContext, Output, op
+    from dagster._core.definitions.decorators.asset_decorator import graph_asset_no_defaults
+    from dagster._core.storage.asset_check_execution_record import AssetCheckExecutionRecordStatus
+
+    check_specs = []
+    for c in checks:
+        check_specs.extend(c.specs)
+
+    check_output_names = [c.get_python_identifier() for c in check_specs]
+
+    check.invariant(len(asset_def.op.output_defs) == 1)
+    asset_out_type = asset_def.op.output_defs[0].dagster_type
+
+    @op(ins={"materialization": In(asset_out_type), "check_evaluations": In(Nothing)})
+    def fan_in_checks_and_materialization(context: OpExecutionContext, materialization):
+        yield Output(materialization)
+
+        for check_spec in check_specs:
+            executions = context.instance.event_log_storage.get_asset_check_executions(
+                asset_key=asset_def.key, check_name=check_spec.name, limit=1
+            )
+            check.invariant(
+                len(executions) == 1, "Expected asset check {check_spec.name} to execute"
+            )
+            execution = executions[0]
+            check.invariant(
+                execution.run_id == context.run_id,
+                "Expected asset check {check_spec.name} to execute in the current run",
+            )
+            if execution.status != AssetCheckExecutionRecordStatus.SUCCEEDED:
+                raise DagsterAssetCheckFailedError()
+
+    def blocking_asset(**kwargs):
```

Comment:
do we not know what the arguments are ahead of time? If so, let's break it out. If we do not, leave a comment as to what these represent.

Reply:
they're the inputs to the passed in asset op. commented